### ISSUE TITLE
Bug: eslint cant load esm plugin

### ISSUE THREAD

**Issue by simlu** [eslint/eslint#16198]
### Environment

Node version: 
npm version: 16.15.0
Local ESLint version: 8.21.0
Global ESLint version: -
Operating System: Ubuntu

### What parser are you using?

Default (Espree)

### What did you do?

Trying to load an esm plugin with `new ESLint` and I can't get it to work. Below should work according to documentation [here](https://github.com/eslint/rfcs/pull/40/files#diff-350c9b4a729ce6a1b3b60eafcd5f38f3b27ea4b9b4b7523e5e360b59afd0a117R55).

<details>
<summary>Minimal Example</summary>

```js
import { dirname } from 'path';
import { fileURLToPath } from 'url';
import { ESLint } from "eslint";
import bf from '@blackflux/eslint-plugin-rules';
import json from "eslint-plugin-json";
import mocha from "eslint-plugin-mocha";

const eslint = new ESLint({
  cwd: dirname(fileURLToPath(import.meta.url)),
  fix: false,
  baseConfig: {
    root: true,
    extends: [
      'airbnb-base',
      'plugin:mocha/recommended',
      'plugin:markdown/recommended'
    ],
    rules: {
      '@blackflux/rules/c8-prevent-ignore': 1
    },
    env: {
      es6: true,
      node: true,
      mocha: true
    },
    globals: {},
    parser: '@babel/eslint-parser',
    parserOptions: {
      requireConfigFile: false
    },
    plugins: [
      { id: '@blackflux/rules', definition: bf }
    ]
  },
  // we use glob on passed in files, due to https://github.com/eslint/eslint/issues/5623
  ignore: false,
  reportUnusedDisableDirectives: 'error'
});

const results = await eslint.lintFiles([
  fileURLToPath(import.meta.url)
]);

```
</details>

But I'm getting `The "path" argument must be of type string. Received an instance of Object`

### What did you expect to happen?

Eslint should load the plugin

### What actually happened?

Eslint can't load the plugin

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by harish-sethuraman** [comment]
I suppose this should be a problem?

``` js
plugins: [
      { id: '@blackflux/rules', definition: bf }
    ]
```

**Comment by ljharb** [comment]
This isn't a bug; eslint uses `require`, and you can't require ESM. Additionally, ESM is async, and eslint plugins/configs must be sync.

**Comment by simlu** [comment]
> This isn't a bug; eslint uses `require`, and you can't require ESM. Additionally, ESM is async, and eslint plugins/configs must be sync.

According to this ticket you can though?
https://github.com/eslint/eslint/issues/15453#issuecomment-1001200953

**Comment by ljharb** [comment]
No, this comment clarifies that you can’t. https://github.com/eslint/eslint/issues/15453#issuecomment-1002015088

**Comment by simlu** [comment]
> No, this comment clarifies that you can’t. [#15453 (comment)](https://github.com/eslint/eslint/issues/15453#issuecomment-1002015088)

I was talking about the comment after the one you linked. The last one in the ticket! It details that you can when using eslint programmatically

**Comment by ljharb** [comment]
I suppose that’s true, altho I’m not sure why you’d want to do that.

**Comment by mdjermanovic** [comment]
@simlu the latest API documentation is here:

https://eslint.org/docs/latest/developer-guide/nodejs-api

You can use `options.plugins` to pass in the plugin implementation.

```js
import { ESLint } from "eslint";
import bf from '@blackflux/eslint-plugin-rules';

const eslint = new ESLint({
  plugins: {
    '@blackflux/rules': bf
  },
  baseConfig: {
    plugins: [
      '@blackflux/rules'
    ]
    // ...
  },
  // ...
});
```

**Comment by simlu** [comment]
Perfect, this seems to work well

[Reference PR](https://github.com/blackflux/js-gardener/pull/1696/files)

Closing this, thank you all for your input!
